core/admin.py

```diff
@@ -12,36 +12,46 @@
 from . import tasks
 
 
-def _text_preview(log_file: FileField):
+def _text_preview(target_file: FileField, mimetype, show_end):
     """
     Return the text of a file if it is short or the last portion of it if it is long.
 
-    :params: log_file A FileField to read text from.
+    params: target_file A FileField to read text from.
+            mimetype Log or Data's mimetype, score result don't have one
+            show_end Determine if showing the first/last portion for preview
     """
+    mimetype_check = mimetype.startswith('text/')
     # max file size for display, currently 10kb
     maxlen = 10000
-    if log_file:
-        with log_file.open('rb') as datafile:
+    if target_file:
+        with target_file.open('rb') as datafile:
             if len(datafile) > 0:
-                try:
+                if show_end:
+                    # read and only display from the end for log filess                    display_message = 'last'
                     datafile.seek(-maxlen, os.SEEK_END)
-                except OSError as exc:
-                    if exc.errno != 22:
-                        # reraise exceptions except for trying to seek before the beginning
-                        raise
-                message = datafile.read().decode(errors='replace')
-                if len(log_file) < maxlen:
+                    message = datafile.read().decode(errors='replace')
```

Comment:
We can have a single read call for both conditions of `show_end` (that is, `datafile.read(maxlen).decode(errors='replace')` is valid for both conditions), so the read can be outside of the `if`/`else`.